src/components/buttons/ScrollTopButton.tsx

```diff
@@ -0,0 +1,40 @@
+import arrowTop from '../../../public/ikony/scroll-top.svg?url';
+import arrowTopHover from '../../../public/ikony/scroll-top-hover.svg?url';
+import { useState } from 'react';
+
+const ScrollToTopButton = () => {
+  const [isHovering, setIsHovering] = useState(false);
+  const scrollToTop = () => {
+    window.scrollTo({ top: 0, behavior: 'smooth' });
+  };
+
+  const baseStyle = {
+    borderColor: 'orange',
+    transition: 'background-color 0.2s ease',
+  };
+  
+  const hoverStyle = {
+    ...baseStyle,
+    backgroundColor: 'orange', 
+  };
+
+  return (
+    <div>
+      <button
+        onClick={scrollToTop}
+        onMouseEnter={() => setIsHovering(true)}
+        onMouseLeave={() => setIsHovering(false)}
+        style={isHovering ? hoverStyle : baseStyle}
```

Comment:
instead of using javascript. it could be handled via pseudo selectors 
`className="border-orange transition-colors ease hover:bg-orange"`

https://tailwindcss.com/docs/transition-property

so then you can get rid of the onMouseEnter, .. useState and the style objects